added #BETWEEN syntax to SQL

EdoFede opened the pull request:
This pull request introduces a new feature to handle SQL `#BETWEEN` expressions in the `gnrpy/gnr/sql/gnrsqldata.py` file. The most important changes include adding a new regular expression for detecting `#BETWEEN` expressions, updating various methods to support this new feature, and implementing the logic for expanding `#BETWEEN` expressions.

Enhancements to SQL expression handling:

* Added `BETWEENFINDER` regular expression to detect `#BETWEEN` expressions in SQL statements. (`[gnrpy/gnr/sql/gnrsqldata.pyR54](diffhunk://#diff-40b2ac58f933a67c5b353728dadced365323547a541bb587c822430a0cae7d24R54)`)
* Updated the `expandEnv` method to include the expansion of `#BETWEEN` expressions using the `BETWEENFINDER` regular expression. (`[gnrpy/gnr/sql/gnrsqldata.pyR250](diffhunk://#diff-40b2ac58f933a67c5b353728dadced365323547a541bb587c822430a0cae7d24R250)`)
* Modified `_getRelationAlias` method to support the new `#BETWEEN` expressions and added a TODO comment to deprecate the old `between` handling. (`[gnrpy/gnr/sql/gnrsqldata.pyR361-R364](diffhunk://#diff-40b2ac58f933a67c5b353728dadced365323547a541bb587c822430a0cae7d24R361-R364)`)
* Updated the `compiledQuery` method to handle `#BETWEEN` expressions in the `where` clause. (`[gnrpy/gnr/sql/gnrsqldata.pyR606](diffhunk://#diff-40b2ac58f933a67c5b353728dadced365323547a541bb587c822430a0cae7d24R606)`)

Implementation of `#BETWEEN` expansion logic:

* Implemented the `expandBetween` method to convert `#BETWEEN` expressions into SQL conditions. (`[gnrpy/gnr/sql/gnrsqldata.pyR827-R842](diffhunk://#diff-40b2ac58f933a67c5b353728dadced365323547a541bb587c822430a0cae7d24R827-R842)`)

EdoFede committed:
added #BETWEEN syntax to SQL

EdoFede committed:
fix: adjust SQL range comparison to be inclusive of upper bound

cgabriel committed:
Merge branch 'develop' into feature/between_finder

cgabriel commented:
I'm adding tests coverage before reviewing it.

cgabriel committed:
added test case for #BETWEEN usage in where conditions

cgabriel commented:
We'll need a new db structure XML dump to cover the feature with tests, when BETWEEN is being used in a relation description

cgabriel committed:
more readable xml data

cgabriel committed:
explore more test cases for BETWEEN, using dates and integers intervals

EdoFede committed:
Add location table definition to SQL common tests

EdoFede committed:
Switch to package-based structure definition
Enhance #BETWEEN expansion tests for blank handling and integer (positive and negative) parameters

cgabriel committed:
added minimanl instanceconfig with framework only package for unit testing

EdoFede committed:
Add location table definition to video package and update XML structure

EdoFede commented:
@cgabriel  Adjusted the test data that caused fails during test sequence.

### New `location` Table:
* Modified the expected table keys in the test to include `location` (`gnrpy/tests/sql/b_structure_build_test.py`).
* Updated also the database structure in legacy XML format to include the new `location` table (`gnrpy/tests/sql/data/dbstructure_base.xml`). [[1]](diffhunk://#diff-dea542b4ed9d566f04ab39b8e1eaf55ad83d70d9cbdb0c2ae8bcada60f670b3bL2-R57) [[2]](diffhunk://#diff-2c872776dc60abdf6f0c4e719f102419a57752e8ba5f75e57997b807b96a98bfL2-R62)

### Test Adjustments:
* Replaced loading of the XML structure with the `configurePackage` function in the query test setup (`gnrpy/tests/sql/e_query_test.py`). [[1]](diffhunk://#diff-35371f1f1af17770004e51215498bf48988df9784caad987f0fc73a289b959f8L40-R40) [[2]](diffhunk://#diff-35371f1f1af17770004e51215498bf48988df9784caad987f0fc73a289b959f8L51-R52)
* Cleaned up unnecessary blank lines in the query test file (`gnrpy/tests/sql/e_query_test.py`).

cgabriel committed:
Merge branch 'develop' into feature/between_finder

cgabriel committed:
the file get generated, so syntax formatting is useless, it should be saved
already formatted.
